src/services/bonusService.ts

```diff
@@ -72,6 +75,15 @@ export default class BonusService {
     | IResponseSuccessRedirectToResource<InstanceId, InstanceId>
   > =>
     withCatchAsInternalError(async () => {
+      // check if current logged user is enabled to start a bonus request
+      if (
+        fromNullable(user.date_of_birth).exists(
+          _ => !isOlderThan(18)(new Date(_), new Date())
+        )
+      ) {
+        return ResponseErrorUnexpectedAuthProblem();
```

Comment:
this returns 500. didn't we say 401 ? (can you add a test for this case ?)

Reply:
replaced 500 with 401 and test added ( similar to previous but I've changed the kind of response expected )